include/miral/miral/window_manager_tools.h

```diff
@@ -39,229 +39,273 @@ struct ApplicationInfo;
- * Workspace is intentionally opaque in the miral API. Its only purpose is to
- * provide a shared_ptr which is used as an identifier.
- *
- * The MirAL implementation of workspaces only prescribes the following:
- *  o When child windows are created they are added to all(any) workspaces of parent
- *  o Focus changes will first try windows with a common workspace
- *  o Adding/removing windows to a workspace affects the whole ancestor/decendent tree
- *
- * The presentation of workspaces is left entirely to the policy
- */
 class Workspace;
 
 class WindowManagerToolsImplementation;
 
-/// Window management functions for querying and updating MirAL's model
+/// This class provides compositor authors with facilities to request state and changes
+/// from windows, applications, workspaces, and more.
+///
+/// When a #WindowManagementPolicy is added to a #miral::MirRunner via #miral::add_window_manager_policy,
+/// the policy will receive an instance of #miral::WindowManagerTools as the first parameter.
+/// Compositor authors are encouraged to use this class instance to handle and request changes from
+/// the underlying Mir server.
 class WindowManagerTools
 {
 public:
+    /// Constructs an instance of window manager tools from an implementation.
+    ///
+    /// Compositor authors are not expected to construct this object themselves.
+    /// Instead, they should use #miral::add_window_manager_policy to create
+    /// an instance of this class for them. For this reason,
+    /// #miral::WindowManagerToolsImplementation is purposefully opaque.
     explicit WindowManagerTools(WindowManagerToolsImplementation* tools);
     WindowManagerTools(WindowManagerTools const&);
     WindowManagerTools& operator=(WindowManagerTools const&);
     ~WindowManagerTools();
 
-/** @name Query & Update Model
- *  These functions assume that the BasicWindowManager data structures can be accessed freely.
- *  I.e. they should only be used by a thread that has called the WindowManagementPolicy methods
- *  (where any necessary locks are held) or via a invoke_under_lock() callback.
- *  @{ */
-
-    /** count the applications
-     *
-     * @return number of applications
-     */
+    /// Returns the number of applications.
+    ///
+    /// @returns the number of applications
     auto count_applications() const -> unsigned int;
 
-    /** execute functor for each application
-     *
-     * @param functor the functor
-     */
+    /// Execute a functor for each application.
+    ///
+    /// \param functor called for each application
     void for_each_application(std::function<void(ApplicationInfo& info)> const& functor);
 
     /** find an application meeting the predicate
      *
      * @param predicate the predicate
      * @return          the application
      */
+    /// Find an application that meets the predicate.
+    ///
+    /// \param predicate the predicate
+    /// \returns an application, or the default-constructed application if none is found
```

Comment:
Don't tell the reader how the result is constructed, tell them the required properties
```suggestion
    /// \returns an application, or a null application if none is found
```